tests/gpflow/test_base.py

```diff
@@ -55,3 +56,24 @@ def test_cast_to_dtype_precision_issue():
     assert actual_value.dtype == np.float64
     expected_value = np.float64(0.2)
     assert actual_value == expected_value
+
+
+def test_parameter_saved():
```

Comment:
can you also check that the number of trainable parameters is the same in the original and reloaded model. I've had issues with that before.

Reply:
The exec returns the same result. If it were a different parameter the result would be different? Or do you mean something else?

Reply:
If you run `model.variables` on both models do you get the same number of parameters back? There used to be an issue that certain variables / parameters were dropped even though the execution of `exec` gave the same result.

Reply:
As far as I understand, `tf.save_model` saves models for deployment purposes only (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md#builder). `m1` can be a different object that wraps graph computation.

Reply:
Keras uses `tf.save_model` format in a different way. Standard `tf.save_model` return not the same object that you pass in at saving time.

```python
> m1
<tensorflow.python.saved_model.load.Loader._recreate_base_user_object.<locals>._UserObject object at 0x13f553fd0>

> m1.variables
AttributeError: '_UserObject' object has no attribute 'variables'
```